assets/javascripts/discourse/components/ai-search-discoveries.gjs

```diff
@@ -73,10 +73,12 @@ export default class AiSearchDiscoveries extends Component {
 
     withPluginApi((api) => {
       api.addSearchMenuOnKeyDownCallback((searchMenu, event) => {
-        if (!searchMenu || this.discobotDiscoveries.loadingDiscoveries) {
+        if (!searchMenu) {
           return;
         }
 
+        this.search.handleArrowUpOrDown(event);
```

Comment:
If I read the core api correctly, this shouldn't be necessary if the callbacks below all return `true`. We only need this because returning `false` somewhere here will skip the default core event handling (for arrows, for Enter, for Escape).

https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/components/search-menu/search-term.gjs#L59-L68

We could do `return true` on each case where we return here, haven't tried it, but I suspect that should allow core's handling to work as expected.